### ISSUE TITLE
[META] Determining resource tag hierarchy

### ISSUE THREAD

**Issue by VeckoTheGecko** [ClimateTown/knowledge-hub#75]
**Describe your suggested improvement**
Decide on a hierarchy of tags/tag guidelines that effectively categorise the resources on the knowledge hub.

**Describe the benefit**
Helps with organisation of the knowledge hub.

**Would you be capable/willing to implement the improvement?**
This is an open discussion with the community. Chime in!

**Additional comments**
Use of `/` within tags to determine specificity (eg. `Politics/UK`) would be good as it is clear, and it gets sorted nicely alphabetically.

**Comment by VeckoTheGecko** [comment]
Sorry about the late response on this @Morzaram. I don't think that this approach would be good for the knowledge hub. The main goal is to have all these resources browsable through a central filtering system (ie. on a single page) so that people can set a filter and scroll through all the related resources at once. This adds more customisability than if things were fractured into separate pages.

Any improvements to navigation will need to come from improvements to the filtering system.

> I think it'll be a lot more sharable if we go with the page route.

Filter settings can still be shared by copying and pasting the URL (if the filter settings are stored as a [query string](https://en.wikipedia.org/wiki/Query_string) in the URL bar). Not sure if the current implementation has this, but it can definitely be a future focus.

Shareability at this point is a bit of a double edged sword at this early stage in the process since it would make tags less dynamic (especially with subpages, as this would involve handling 404s and thats a headache I don't want). I would rather start with flexibility in tag names to start with, and then perhaps include some shareability by implementing query strings to populate filter settings. 

Of course, I really appreciate the suggestion and the thought/discussion behind this 😄 .  This issue is a bit of a low priority until after release and we have more resources in the system

**Comment by dmlb** [comment]
Keeping the structure flat as all filters could get overwhelming depending on how many get added.

 Scrolling through 100 filters is not a great experience and likely surpass the limit of most short term memory from beginning to end of the list. 
https://www.w3.org/TR/coga-usable/user_stories.html#distractions-user-story

to keep the list manageable could collapse some of the, together like workers rights and unions, but keep the other label as a synonym if you plan on building search by query.

How do you see people using it? Casual browsing or determined searching?

the category structure suggested if you have an external search based goal might help, keyword search “climate policy uk” could drive traffic to that hub category.

**Comment by VeckoTheGecko** [comment]
> Scrolling through 100 filters is not a great experience

I agree. We'll be keeping the number of tags down (and, in future, having a specified list of tags that is only added to if a needed resource really doesn't fit into any other category). Also looking at writing descriptions for tags to explain clearly what different tags do. Honestly, I don't see the tag list growing to an unmanageable state as I feel the current tag list is quite encompassing.

> could collapse some of them together like workers rights and unions

I think that's a good change to consolidate those.


> How do you see people using it? Casual browsing or determined searching?

The goal is both casual browsing and determined searching, which is up to the user and how they use the available filters. Features such as "AND" and "OR" mode when selecting tags, and basic search functionality would be really helpful for this.

> Keeping the structure flat as all filters could get overwhelming depending on how many get added.

Yes, I think so too (that for wide filters the resource list can be a bit much). I'm not sure if there is a solution to this except for the user to use the tag list. The bigger problem also becomes "how do you sort the resources" when the list grows long. I was thinking of using analytics to track clicks to all resources, and use that to order resources (ie. popular near the top, sprinkled in with "up and coming". Although thats a long term idea that will only become a priority if/when resources gets >200 I think.


> the category structure suggested if you have an external search based goal might help, keyword search “climate policy uk” could drive traffic to that hub category.

I never envisioned that the Knowledge Hub would show up for general climate Google searches. Moreso thought of it spreading by word of mouth, and for search terms like:

* Climate resources
* Climate knowledge hub
* Climate town knowledge hub

The way the site is at the moment I can imagine is pretty bad for SEO. Since its a single page application with no other pages, but also because of the nature of the site having short descriptions and intentionally linking to other places (reducing time spent on the site).

I don't think its something worth fixing. I think the current approach is on track for:

* The best user experience
* Better maintainability
* A bigger focus on the resources themselves (and not the site)

I don't want to make large changes to the site to appease the SEO gods.

**Comment by JarrodBaniqued** [comment]
> The bigger problem also becomes "how do you sort the resources" when the list grows long. I was thinking of using analytics to track clicks to all resources, and use that to order resources (ie. popular near the top, sprinkled in with "up and coming". Although thats a long term idea that will only become a priority if/when resources gets >200 I think.

The simple and quick option is to just alphabetize them for now. Consider it carefully, though.

**Comment by VeckoTheGecko** [comment]
Interesting comment. Not sure how tags are sorted currently (though alphabetically would help group `Politics/*` and `Location/*` together, so that is a good option.

Sorting by traffic was mainly concerning the resources themselves. For tags, sorting alphabetically sounds good from an organisation standpoint.

**Comment by VeckoTheGecko** [comment]
> I would also suggest a global, all-resource "A-to-Z" sorting option ignoring tags, for those who have the time and want to scroll through the whole list for fun.

Just re-reading this, and noticed it also discusses A-Z sorting of resources.

The easiest way to do this is to just sort the while resource list alphabetically (which I'm opposed to doing, as I think reverse chronological is better (discussed in #130)).

Having options to sort by popularity, time posted, or A-Z would be good, but not a priority given other issues in the site (and I don't know if it will end up being in scope). CSV export funtionality (#131) is definitely something that would suit the data lovers who want to further explore.
